benches/benchmarks.rs

```diff
@@ -70,30 +126,42 @@ fn load_standard_lib(bencher: divan::Bencher) {
 }
 
 #[divan::bench_group]
-mod eval_commands {
+mod record {
+
     use super::*;
 
-    fn bench_command(bencher: divan::Bencher, scaled_command: String) {
-        let mut engine = setup_engine();
-        load_standard_library(&mut engine).unwrap();
-        let commands = Spanned {
-            span: Span::unknown(),
-            item: scaled_command,
-        };
+    fn create_record_string(n: i32) -> String {
+        let mut s = String::from("let record = {");
+        for i in 0..n {
+            s.push_str(&format!("col_{}: {}", i, i));
+            if i < n - 1 {
+                s.push_str(", ");
+            }
+        }
+        s.push('}');
+        s
+    }
 
-        bencher
-            .with_inputs(|| engine.clone())
-            .bench_values(|mut engine| {
-                evaluate_commands(
-                    &commands,
-                    &mut engine,
-                    &mut nu_protocol::engine::Stack::new(),
-                    PipelineData::empty(),
-                    None,
-                )
-                .unwrap();
-            })
+    #[divan::bench(args = [1, 10, 100, 1000])]
+    fn create(bencher: divan::Bencher, n: i32) {
+        bench_command(bencher, create_record_string(n));
```

Comment:
This benchmark is probably dominated by the parsing of the literal and the inner values over the mere cost of insertion into a single record (single `Record` created in the parse/eval step effectively makes the `Box` indirection irrelevant). We can keep it separately to see if our anti-duplication and parsing logic is good but this doesn't answer the question your change to `Value` poses.